app/questionnaire/router.py

```diff
@@ -154,15 +153,10 @@ def get_section_resume_url(self, routing_path):
 
         return self.get_first_location_in_section(routing_path).url()
 
-    def is_survey_complete(self):
+    @cached_property
```

Comment:
It seems fine for our current use, but I guess it could happen. I wanted to avoid scenarios in which it would evaluate more than once. Maybe we should look at passing around the value where appropriate.